examples/java/pom.xml

```diff
@@ -61,36 +50,73 @@
       <dependencies>
         <dependency>
           <groupId>org.apache.beam</groupId>
-          <artifactId>beam-runners-direct-java</artifactId>
+          <artifactId>beam-runners-flink_2.10</artifactId>
           <version>${project.version}</version>
           <scope>runtime</scope>
           <optional>true</optional>
         </dependency>
 
         <dependency>
           <groupId>org.apache.beam</groupId>
-          <artifactId>beam-runners-flink_2.10</artifactId>
+          <artifactId>beam-runners-google-cloud-dataflow-java</artifactId>
           <version>${project.version}</version>
           <scope>runtime</scope>
           <optional>true</optional>
         </dependency>
 
         <dependency>
           <groupId>org.apache.beam</groupId>
-          <artifactId>beam-runners-google-cloud-dataflow-java</artifactId>
+          <artifactId>beam-runners-spark</artifactId>
           <version>${project.version}</version>
           <scope>runtime</scope>
           <optional>true</optional>
         </dependency>
 
         <dependency>
+          <groupId>org.apache.spark</groupId>
+          <artifactId>spark-core_2.10</artifactId>
+          <version>${spark.version}</version>
+          <scope>runtime</scope>
+          <optional>true</optional>
+          <exclusions>
+            <exclusion>
+              <groupId>org.slf4j</groupId>
+              <artifactId>jul-to-slf4j</artifactId>
+            </exclusion>
+          </exclusions>
+        </dependency>
+
+        <dependency>
+          <groupId>org.apache.spark</groupId>
+          <artifactId>spark-streaming_2.10</artifactId>
+          <version>${spark.version}</version>
+          <scope>runtime</scope>
+          <optional>true</optional>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <!-- Include SparkRunner and deps with -P spark-runner -->
+    <profile>
+      <id>spark-runner</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```

Comment:
Done. Specifying a profile deactivates all profiles that are activated due to `activeByDefault`. It doesn't mess up other profiles.